Automattic/wp-calypso
# Domains: Change auth code component button message for the transfer flow

#### Changes proposed in this Pull Request
Wraps the auth code step for the "Use a domain I own" flow and creates a specialized version of the component for both Transfer and Connect options. 
_Also does some TypeScript conversion in a few places so it's easier to prevent errors and validate types more assertively. 😬_

#### Testing instructions
- Same steps as #56178. Additionally, when choosing the Transfer option:

  - Make sure that the button message at the ownership verification step for the is "Check readiness for transfer";
  - Confirm that adding the correct auth code adds the transfer to the cart, and an incorrect auth code shows an error message.

- Also, for the Connect option, make sure that the ownership verification step on the domain connection flow still displays the correct button message ("Check my authorization code") and still connects the domain properly 😅.

#### Preview
![image](https://user-images.githubusercontent.com/18705930/133914110-4801a898-5c91-4bd5-959c-2667df54453c.png)

https://user-images.githubusercontent.com/18705930/134104320-ebd511ea-5875-47bd-b3f3-65f48b3b54a1.mov
